wrangler-transform/widgets/Wrangler-transform.json

```diff
@@ -77,6 +104,40 @@
   ],
   "emit-alerts": true,
   "emit-errors": true,
+  "filters": [
+    {
+      "name": "PreconditionValueJEXL",
+      "condition": {
```

Comment:
Can we also clear the existing contents of the SQL precondition when JEXL is selected? As far as I can see the selection change just hides the field but does not clear it. Correct me if I'm wrong.

Reply:
I think this was done by UI because we wanted to retain user input on connections toggle: https://github.com/cdapio/cdap-ui/pull/467.

So it would be good to check with UI team if there is a way to clear existing contents.
As of now, I don't think it is possible.

Reply:
If it's not possible I guess there's not much we can do about it 🤷🏻

Reply:
Is there any specific reason to clear these fields? It makes sense to retain them according to me when we think in terms of user experience, if a user accidentally clicks on the `SQL` radio button after adding filling up the fields related to `JEXL` they would not want to fill them up again after switching back to `JEXL`.

Reply:
The input in the other field will never be used so it would be nice to clear it so that if it is accidentally referenced somewhere in code it immediately throws an exception. We will add validation based on the toggle so clearing the field isn't strictly required, though.

Reply:
I think we should get some product input here.